azkaban-common/src/test/java/azkaban/executor/InteractiveTestJob.java

```diff
@@ -141,4 +142,10 @@ public void cancel() throws InterruptedException {
     info("Killing job");
     failJob();
   }
+
+  public static void clearTestJobs(final String... names) {
```

Comment:
Correct (really well expressed question by the way 👍).

- JobRunner threads only put to `testJobs` without checking it.
- Test main thread is the only thread to call `getTestJob` or `clearTestJobs`.